### DISCUSSION TITLE
Github is flooded with social engineering attacks and nothing can be done about it?

### DISCUSSION THREAD

**Discussion by macromeer** [community/community#136836] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I opened an issue on a repo and immediately got two identical comments asking me to download something and to install it: https://github.com/napari/napari/issues/7213. Why can't I flag comments or accounts for review such as can be done on other platforms?

**Comment by ferid333** [comment]
Yes i have noticed that as well. You can try to contact with support

**Reply by MitchBomcanhao** [reply]
well, you **can** open each commenter's user account and report them through there. it is a bit of a faff, and seeing as these seem like hijacked user accounts, not sure how much benefit there is in pursuing individual accounts.

**Comment by Ammar-Alnagar** [comment]
I got the same problem but i noticed more and more are getting flagged as spam and getting removed 
\

**Comment by konstin** [comment]
There is a huge number of these spam comments, they all seem to be pointing to the same link: https://github.com/search?q=%22https%3A%2F%2Fwww.mediafire.com%2Ffile%2Fo50xaz6wgtazqnx%2Ffix.zip%2Ffile%22&type=issues

**Reply by ebndev** [reply]
@konstin we appreciate you flagging this! I'll go ahead and forward this to be investigated.

**Reply by konstin** [reply]
We're still getting comments: https://github.com/astral-sh/uv/issues/6675#issuecomment-2311782157

**Reply by majora2007** [reply]
I just got hit twice within the past hour. It's the same message pattern, why can't Github detect this pattern and block it. It's extremely annoying and time consuming deleting 20 comments across issues.

**Comment by ebndev** [comment]
Hi @macromeer, 👋🏻. The best route to get this to the proper GitHub team is to use our [abuse reporting tools](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam). Here's all the info:

- [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam)
- [Reporting a user](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-user)
- [Reporting an issue or pull request](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-an-issue-or-pull-request)
- [Reporting a comment](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-comment)

You can report behavior and content that violates community guidelines and terms. We are going to close this post, but for this and any future incidents, please refer to the links above.

Thank you!

**Reply by konstin** [reply]
The reporting-a-user-for-spam form is making me click through a captcha every time is was reporting one of the spam bots, and then the ui also complained that i was reporting too many accounts, even through there were several spam accounts.

**Reply by MitchBomcanhao** [reply]
The worst bit here is that these seem to be legitimate accounts taken over by whatever virus they've installed. Trashing these accounts altogether might have unintended consequences.

**Reply by ebndev** [reply]
@konstin I've flagged this to the proper GitHub team so you don't have to report the users 1 by 1 for this case, sorry for that confusing experience!

**Reply by ThiefMaster** [reply]
Yeah, that reporting form is rather unpleasant to use. It feels exactly like the support ticket form it is. Even the (already convoluted) account reporting process on Instagram is less annoying than this.

**Reply by konstin** [reply]
hey @ebndev, the spammers are back and the reporting process is still a hurdle, it makes me enter a reason and solve a captcha each time, even though the content is obvious bot created porn spam.

**Comment by ThiefMaster** [comment]
It would be great if GitHub had some internal monitoring for such patterns. Many accounts suddenly posting the same link - especially one never seen before - is suspicious and warrants a human to have a look.

**Comment by Aokromes** [comment]
still commenting malware links
https://github.com/telegramdesktop/tdesktop/issues/28322#issuecomment-2311786936

**Comment by jackpoz** [comment]
Just got 4 different users at https://github.com/pnp/powershell/issues/4193 in 5 minutes leaving malware comments. Reported as many as I could before GitHub report system told me I reported too many :P

**Comment by wargio** [comment]
Virustotal run of the passwordless rar file: https://www.virustotal.com/gui/file/f0e6b26c9bbc64d0724108b2c3b23753ee2232d4975a39489a129e419299250e

**Comment by caineblood** [comment]
a simple search of all issues for "password: changme" shows over 4k posts with the same malware in the last 32hrs; some have been deleted but there are *alot* that have yet to be addressed; especially in large/popular gits such as home assistant. https://github.com/search?q=%22password%3A+changeme%22&type=issues

**Reply by DecimalTurn** [reply]
If you want to get only in the last few days, you need to modify your search query like this: https://github.com/search?q=%22password%3A+changeme%22+updated%3A%3E2024-08-25+&type=issues&s=created&o=desc

The number of these is currently going down and I get roughly 300 results.

**Reply by DecimalTurn** [reply]
A lot of the remaining results are people replying to the malicious comment and including the original message in their reply.

**Comment by caineblood** [comment]
After further investigation I have determined that it is automatically replying to new comments; i have had over 30 instances of it immediately replying with *another* malware link the moment i post a warning.

**Comment by caineblood** [comment]
they have now changed from mediafire to a bit.ly link, but the gcc and password part is still included.

**Reply by Aokromes** [reply]
> they have now changed from mediafire to a bit.ly link, but the gcc and password part is still included.

yeah now malware is hosted on discord.

**Comment by ebndev** [comment, marked as answer]
Thanks all for sending updates! The Platform Health team is aware of this issue and are continuing to monitor it.

**Reply by DecimalTurn** [reply]
Thanks for the follow up.

Until this is fully solved, is it a good idea to use a GitHub Action to filter out comments right after they are posted to remove those suspicious links?

**Reply by ebndev** [reply]
@DecimalTurn for your own issues / repositories?

**Reply by DecimalTurn** [reply]
I did it mainly to monitor comments in my own repositories, but I also made the [action](https://github.com/marketplace/actions/comment-filter) available on the marketplace. However, I wanted to confirm that this wouldn't interfere with what the Platform Health team is doing.

Also, I wasn't able to test if the email notification will always be sent with the original link or if the github-action bot is fast enough to change the content before the email notification is sent.

**Reply by ebndev** [reply]
@DecimalTurn that's cool that you did that! Checked with the team and they said it should be fine 👍

**Reply by Aokromes** [reply]
https://github.com/telegramdesktop/tdesktop/issues/28335#issuecomment-2322795369 they are back

**Reply by MikeMcC399** [reply]
@Aokromes The user `KrissV2` seems to have been ~~automatically~~ deleted now.

**Reply by ThiefMaster** [reply]
Yes, I reported `KrissV2` maybe half an hour ago and just got a reply from GitHub that "something" has been done about it.

**Reply by Aokromes** [reply]
> @Aokromes The user `KrissV2` seems to have been ~automatically~ deleted now.

yeah i reported him after write here, i wrote here in case different departament handles this and spam control.

**Reply by CypherpunkSamurai** [reply]
@ebndev the accounts commenting on the issues are legitimate accounts. I think straight away banning them would be wrong. It could have been from personal token leak or password breach.

I think what could be implemented is  a blacklist of links from previous reported comments. If possible with short URL resolver as well to get full URL from links.

To curb the account takeover I think a cooldown and a karma system could be implemented. A user who posts too many links / comments on unrelated issues can have a exponentially increasing cooldown.

It's still commenting
https://github.com/openvax/pyensembl/issues/310

**Comment by ThiefMaster** [comment]
They now switched to spamming issues.

**Comment by SNWCreations** [comment]
This issue still persists. I've also spammed by someone just few hours ago and I've reported that. Looking forward to official updates.
The malware also spreads using Discord CDN and named "fix.zip".
I've tricked by that, sorry for my stupid. It is lucky that I've revoked the attacker's session on my GitHub account. If you are attacked, revoke their session on your account before anything got worse!

**Comment by konstin** [comment]
They are back (https://github.com/astral-sh/uv/issues/7018#issuecomment-2329093177):

![image](https://github.com/user-attachments/assets/827ff7b0-d647-40b3-86d5-5d938c7f3676)
